File: roc_app/Managers/Manager.cpp
#include "stdafx.h"
#include "Managers/Manager.h"

ROC::Manager::Manager(Core *p_core)
{
    m_core = p_core;
    m_started = false;
}

void ROC::Manager::Start()
{
    if(!m_started)
        m_started = true;
}

void ROC::Manager::Stop()
{
    if(m_started)
        m_started = false;
}

bool ROC::Manager::IsActive() const
{
    return m_started;
}

ROC::Core* ROC::Manager::GetCore()
{
    return m_core;
}

File: roc_app/Engine/Managers/Manager.cs
namespace ROC.Engine.Managers
{
    public class Manager
    {
        protected bool m_active = false;

        internal virtual void Start()
        {
            m_active = true;
        }

        internal virtual void Stop()
        {
            m_active = false;
        }

        public bool IsActive => m_active;
    }
}
